### PR TITLE
feat: dockerize api

### LINKED ISSUE #29: Create Dockerfile for API
Create a Dockerfile for the API under `apps/api`. Make sure that you properly use the `ENV` directive to list all the environmental variables the API is using. You can find them from both the code and the `.env.example`

### PR DESCRIPTION
**Pull request by ezhil56x**
## Description
- Added Dockerfile to `apps/api`
- Moved Dockerfile-web to `apps/web` as Dockerfile

Fixes #29

## Future Improvements

- As Dockerfile-web to `apps/web` as Dockerfile, some change are to be made on workflows
- `start:api` should be added to package.json

## Mentions

@rajdip-b 

## Developer's checklist

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-check on my work

**If changes are made in the code:**

- [x] I have followed the [coding guidelines](https://google.github.io/styleguide/jsguide.html)
- [x] My changes in code generate no new warnings
- [ ] My changes are breaking another fix/feature of the project
- [ ] I have added test cases to show that my feature works
- [ ] I have added relevant screenshots in my PR
- [ ] There are no UI/UX issues

### Documentation Update

- [ ] This PR requires an update to the documentation at [docs.keyshade.xyz](https://docs.keyshade.xyz)
- [x] I have made the necessary updates to the documentation, or no documentation changes are required.

### CODE REVIEW

File: apps/api/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,19 @@
+FROM node:20-alpine as build
+
+WORKDIR /app
+
```
Comment (rajdip-b) [inline apps/api/Dockerfile:4]: Please add the environmental variables in here so that it's easier for the users to understand what's being consumed!

**Review by rajdip-b — CHANGES_REQUESTED** [review]
Additionally, do add a `.dockerignore` file in both web and api. for now just add `node_modules` in that file. Later on, we can add more stuff when needed

File: apps/api/Dockerfile (hunk 1 above)
Comment (ezhil56x) [reply apps/api/Dockerfile:21]: Done 👍🏻

**Review by rajdip-b — CHANGES_REQUESTED** [review]
Just make the update to `web.yaml` workflow, and this PR should be good to go!

**Comment by ezhil56x** [comment]
> Just make the update to `web.yaml` workflow, and this PR should be good to go!

Done 👍🏻

**Review by rajdip-b — APPROVED** [review]

**Comment by rajdip-b** [comment]
Made some changes to the final PR:
- remove the comments from Dockerfile. Guess we wont need them afterall, sorry for bothering you with that
- Add the `start:api` script in package.json

@ezhil56x

**Review by rajdip-b — APPROVED** [review]

**Review by rajdip-b — APPROVED** [review]
All looks good now!

**Pull request merged**